services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/security/UserIdLoggingFilter.kt

```diff
@@ -0,0 +1,31 @@
+package fi.hel.haitaton.hanke.security
+
+import jakarta.servlet.Filter
+import jakarta.servlet.FilterChain
+import jakarta.servlet.ServletRequest
+import jakarta.servlet.ServletResponse
+import jakarta.servlet.http.HttpServletRequest
+import jakarta.servlet.http.HttpServletResponse
+import org.slf4j.MDC
+import org.springframework.security.core.context.SecurityContextHolder
+import org.springframework.stereotype.Component
+
+@Component
+class UserIdLoggingFilter : Filter {
```

Comment:
Would it make sense to have a unit test to verify that the userId is logged?

Reply:
I'm not sure how to test this. This would be a controller test, but the integration tests have their own log format. I could maybe add the userId to that format as well if I redefine the pattern, but then I would be testing the test log pattern.

Reply:
Yup, testing a test does not make a lot of sense. I guess there is not much we can do then.